tests/unit/metadata/test_metadata.py

```diff
@@ -129,3 +129,17 @@ def test_dist_found_in_zip(tmp_path: Path) -> None:
     dist = get_environment([location]).get_distribution("pkg")
     assert dist is not None and dist.location is not None
     assert Path(dist.location) == Path(location)
+
+
+@pytest.mark.parametrize(
+    "path",
+    (
+        "/path/to/foo.egg-info".replace("/", os.path.sep),
+        # Tests issue fixed by https://github.com/pypa/pip/pull/2530
+        "/path/to/foo.egg-info/".replace("/", os.path.sep),
+    ),
+)
+def test_trailing_slash_directory_metadata(path: str) -> None:
```

Comment:
This test breaks because we remove `InstallRequirement#get_dist()`, but this test also shouldn't have been done via `InstallRequirement` anyway. So this is a more correct version of the same test, but it's also necessary because of the change to `InstallRequirement` in this diff.